AppCenterReactNativeShared/android/src/main/java/com/microsoft/appcenter/reactnative/shared/AppCenterReactNativeShared.java

```diff
@@ -15,67 +15,51 @@
 
 public class AppCenterReactNativeShared {
 
+    private static final String APPCENTER_CONFIG_ASSET = "appcenter-config.json";
+
     private static final String APP_SECRET_KEY = "app_secret";
 
     private static final String START_AUTOMATICALLY_KEY = "start_automatically";
 
+    private static Application sApplication;
+
     private static String sAppSecret;
 
     private static boolean sStartAutomatically;
 
-    private static Application sApplication;
-
-    private static WrapperSdk sWrapperSdk = new WrapperSdk();
-
     @SuppressWarnings("unused")
-    public static void configureAppCenter(Application application) {
+    public static synchronized void configureAppCenter(Application application) {
         if (sApplication != null) {
             return;
         }
         sApplication = application;
-        sWrapperSdk.setWrapperSdkVersion(com.microsoft.appcenter.reactnative.shared.BuildConfig.VERSION_NAME);
-        sWrapperSdk.setWrapperSdkName(com.microsoft.appcenter.reactnative.shared.BuildConfig.SDK_NAME);
-        AppCenter.setWrapperSdk(sWrapperSdk);
-        if (!AppCenterReactNativeShared.sStartAutomatically) {
+        WrapperSdk wrapperSdk = new WrapperSdk();
+        wrapperSdk.setWrapperSdkVersion(com.microsoft.appcenter.reactnative.shared.BuildConfig.VERSION_NAME);
+        wrapperSdk.setWrapperSdkName(com.microsoft.appcenter.reactnative.shared.BuildConfig.SDK_NAME);
+        AppCenter.setWrapperSdk(wrapperSdk);
+        readConfigurationFile();
+        if (!sStartAutomatically) {
+            AppCenterLog.debug(LOG_TAG, "Configure not to start automatically.");
             return;
         }
 
         /* Get app secret from appcenter-config.json file. */
-        String appSecret = getAppSecret();
```

Comment:
That's where the bug was, we reading the file after reading `sStartAutomatically`, chicken/egg issue. iOS code was doing that in the right order.